utils/language.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2013-2020 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const osLocale = require('os-locale');
+const { LANGUAGES } = require('../generators/generator-constants');
+
+function detectLanguage() {
+    const locale = osLocale.sync();
```

Comment:
With my computer (using Ubuntu 18.04), the value is: `locale=fr-FR`
So it can't match to `fr`

Reply:
We could fall back removing `-FR`.
But looks like java recommends using `fr_FR` instead.
https://www.oracle.com/java/technologies/javase/jdk8-jre8-suported-locales.html

Looks like for each `xx` there is a `xx_XX`:
https://www.viralpatel.net/java-locale-list-tutorial/

Our current localeId is angular locale and looks less useful, maybe is best to remove:
https://github.com/jhipster/generator-jhipster/blob/c8868be7e3b3ccd07201e01f9ae4d8f72308fb83/generators/generator-constants.js#L275-L287